satellite/notification/remotelogger.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notification
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"storj.io/storj/pkg/sap"
+	"storj.io/storj/pkg/storj"
+
+	"storj.io/storj/pkg/pb"
+)
+
+type RemoteLogger struct {
+	zap.Logger
+	target storj.NodeID
+	*Service
+}
+
+func (log *RemoteLogger) Remote(target storj.NodeID) sap.Logger {
+	return log
+}
+
+func (log *RemoteLogger) Named(s string) sap.Logger {
+	return &RemoteLogger{*log.Logger.Named(s), log.target, log.Service}
+}
+
+func (log *RemoteLogger) ProcessDebug(message string, fields ...zapcore.Field) {
+	log.processNotification(message, pb.LogLevel_DEBUG)
+}
+
+func (log *RemoteLogger) ProcessInfo(message string, fields ...zapcore.Field) {
+	log.processNotification(message, pb.LogLevel_INFO)
+}
+
+func (log *RemoteLogger) ProcessWarn(message string, fields ...zapcore.Field) {
+	log.processNotification(message, pb.LogLevel_WARN)
+}
+
+func (log *RemoteLogger) ProcessError(message string, fields ...zapcore.Field) {
+	log.processNotification(message, pb.LogLevel_ERROR)
+}
```

Comment:
The renaming violates the sap.Logger interface, which idea is to be compliant with the zap.Logger. to be able replace the logger where necessary without friction.